api/models/environment.go

```diff
@@ -15,24 +15,51 @@ import (
 
 type Environment map[string]string
 
-func LoadEnvironment(data []byte) Environment {
+// cleanEnvPair validates environment variable keypair format, trims spaces and surrounding single quotes.
+func cleanEnvPair(value string) (string, error) {
+	parts := strings.SplitN(value, "=", 2)
+	if len(parts) != 2 {
+		return "", fmt.Errorf("Environment variables should be defined in key=value format. You specified: " + value)
+	}
+
+	if key := strings.TrimSpace(parts[0]); key != "" {
+		val := parts[1]
+
+		// heroku env -s adds leading and trailing single quotes to val. Strip.
+		val = strings.Trim(val, "'")
+		val = strings.TrimSpace(val)
+
+		if len(val) == 0 {
+			return "", fmt.Errorf("Can't set " + key + " to an empty value; try `convox env unset`.")
```

Comment:
I think you should be able to set env vars to an empty value. If the docker-compose.yml list of env vars is a "manifest" of required vars then some people will need to set some of them to blank.